crates/trigger-http/src/lib.rs

```diff
@@ -573,13 +600,22 @@ pub(crate) trait HttpExecutor: Clone + Send + Sync + 'static {
         route_match: &RouteMatch,
         req: Request<Body>,
         client_addr: SocketAddr,
+        self_definition: SelfRequestDefinition,
     ) -> Result<Response<Body>>;
 }
 
+/// The definition of the `self` host for outbound requests.
+#[derive(Clone)]
+pub struct SelfRequestDefinition {
```

Comment:
Could this perhaps be `SelfRequestOrigin`? "Definition" is pretty vague and I doubt future readers would have a reasonable intuition for what it stands for.

The `origin` concept of course technically includes the port, too; one way to address that would be to simply include it. But I think it'd also be fine to just add a comment saying that for our purposes that part is irrelevant, so left out.

Reply:
`Authority` does include port, either explicitly or implicitly via scheme. This struct could also alternatively wrap (or be replaced by) a `Uri`.

Reply:
I renamed to `SelfRequestOrigin` and kept its fields (`URI` contains a lot of other stuff we don't need and I think it makes the code harder to read).